### PR TITLE
Repository name in cached files

### LINKED ISSUE #6: Option to download the files that have hits
It would be nice to be able to download the files that have search hits so you can open them locally.

### PR DESCRIPTION
**Pull request by mpopdam**
Fixes #6

### CODE REVIEW

**Comment by mpopdam** [comment]
@martijnspaan Could you review this PR?

File: source/FileCache.cs [hunk 1]
```diff
@@ -8,28 +7,28 @@ namespace GitHubSearch
...
-        public FileCache()
+        public FileCache(IFileSystem fileSystem)
         {
-            if (!Directory.Exists(LocalCacheFolder))
-            {
-                Directory.CreateDirectory(LocalCacheFolder);
-            }
+            this.fileSystem = fileSystem;
+
+            fileSystem.EnsureDirectoryEsists(LocalCacheFolder);
```
Comment (martijnspaan) [inline source/FileCache.cs]: Typo: Exists

**Review by martijnspaan — COMMENTED** [review]
Review complete. Also please rebase

File: source/FileCache.cs (hunk 1 above)
Comment (mpopdam) [reply source/FileCache.cs]: Fixed it

**Pull request merged**